src/utils/PromptCreator.ts

```diff
@@ -6,7 +6,36 @@ import {
 } from "@/types";
 
 export class PromptCreator {
-    constructor(private apiCatalog: TestingFrameworkAPICatalog) {}
+    constructor(private apiCatalog: TestingFrameworkAPICatalog) {
```

Comment:
Both the constructor and `extendAPICategories` are doing the same operation - merging categories based on their titles. We can extract this shared logic and use it within one util method.

Reply:
e.g.

```ts
    private mergeCategories(categories: TestingFrameworkAPICatalogCategory[]): void {
        const categoryMap = this.apiCatalog.categories.reduce((acc, category) => {
            acc.set(category.title, category);
            return acc;
        }, new Map<string, TestingFrameworkAPICatalogCategory>());

        categories.forEach(category => {
            const existing = categoryMap.get(category.title);
            if (existing) {
                existing.items.push(...category.items);
            } else {
                categoryMap.set(category.title, {...category});
            }
        });

        this.apiCatalog.categories = Array.from(categoryMap.values());
    }
```

Reply:
also we can use more functional pattern - [reduce](https://geeksforgeeks.org/typescript-array-reduce-method/):
```ts
private mergeCategories(categories: TestingFrameworkAPICatalogCategory[]): TestingFrameworkAPICatalogCategory[] {
    return categories.reduce((mergedCategories, category) => {
        const existingIndex = mergedCategories.findIndex(c => c.title === category.title);
        
        if (existingIndex >= 0) {
            mergedCategories[existingIndex].items.push(...category.items);
            return mergedCategories;
        }

        return [...mergedCategories, {...category}];
    }, [] as TestingFrameworkAPICatalogCategory[]);
}
```

Reply:
if category doenst exist then existingIndex will be -1 (that's how findIndex works)

Reply:
then, we can call the same method this way:

```ts
constructor(private apiCatalog: TestingFrameworkAPICatalog) {
    this.mergeCategories(this.apiCatalog.categories);
}

extendAPICategories(
    newCategories: TestingFrameworkAPICatalogCategory[]
): void {
    this.apiCatalog.categories = this.mergeCategories([...this.apiCatalog.categories, ...newCategories]);
}
```

Reply:
also I removed the option of one category from my example as we said we'll change it to only array

Reply:
Now, that we have a shared logic in a common function - we can maintain it from one place and improve it without duplicating.
For example - what happens if someone accidently calls the extend function with the same parameters over and over again (e.g. decided to put that on the `beforeEach` which is called multiple time)? We want to avoid redundancy and duplications in the new catalog. 

So we can warn / err them or just avoid the duplications, to do that, we can use the old trick of transforming the array of items to set → back to array, e.g.:

```ts
const uniqueItems = Array.from(new Set(category.items));
```